### PR TITLE
LIME-41: Add header component

### LINKED ISSUE #41: Add Header component
Image

### PR DESCRIPTION
**Pull request by ArtemLymarenko**
Added header component.

### CODE REVIEW

**Review by valery-chumak — APPROVED** [review]

**Comment by ArtemLymarenko** [comment]
Result

изображение

File: frontend/src/bundles/common/components/header/header.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import { getValidClassNames } from '../../helpers/helpers.js';
+import { Layout, Link } from '../components.js';
+import { Message, Navigation } from './components/components.js';
+import styles from './styles.module.css';
+
+const Header = (): JSX.Element => {
+    return (
+        <header className={styles['header']}>
+            <Layout className={getValidClassNames(styles['header-container'])}>
```
Comment (SofiiaTrokhymchuk) [inline frontend/src/bundles/common/components/header/header.tsx]: Why `getValidClassNames` is needed here? As far as I see you also pass below in the code single class using `styles[some-class]` but without this helper function

Comment (ArtemLymarenko) [reply frontend/src/bundles/common/components/header/header.tsx]: Thank you for noticing. I'll fix this.

File: frontend/src/bundles/common/components/header/components/navigation/navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { BellIcon, Cog6ToothIcon } from '@heroicons/react/24/solid';
+
+import defaultAvatar from '~/assets/img/default-avatar.svg';
+import { Link } from '~/bundles/common/components/components.js';
+
+import styles from './styles.module.css';
+
+type Properties = {
+    avatarUrl: string | null;
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/header/components/navigation/navigation.tsx]:
```suggestion
    avatarUrl?: string;
```

File: frontend/src/bundles/common/components/header/components/navigation/navigation.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+const Navigation = ({ avatarUrl }: Properties): JSX.Element => {
+    return (
+        <nav className={styles['navigation']}>
+            <ul className={styles['menu-list']}>
+                <li>
+                    <Link to={'/'}>
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/header/components/navigation/navigation.tsx]: We have this path in AppRoute object

File: frontend/src/bundles/common/components/header/components/navigation/navigation.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                        <BellIcon className={styles['icon']} />
+                    </Link>
+                </li>
+                <li>
+                    <Link to={'/'}>
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/header/components/navigation/navigation.tsx]: same here

File: frontend/src/bundles/common/components/header/components/navigation/navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                        <Cog6ToothIcon className={styles['icon']} />
+                    </Link>
+                </li>
+                <li>
+                    <Link to={'/'}>
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/header/components/navigation/navigation.tsx]: and here

File: frontend/src/bundles/common/components/header/header.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import { Layout, Link } from '../components.js';
+import { Message, Navigation } from './components/components.js';
+import styles from './styles.module.css';
+
+const Header = (): JSX.Element => {
+    return (
+        <header className={styles['header']}>
+            <Layout className={`${styles['header-container']}`}>
+                <div className={styles['header__logo-wrapper']}>
+                    <Link to={'/'}>
```
Comment (anton-otroshchenko) [inline frontend/src/bundles/common/components/header/header.tsx]: and also here

**Review by anton-otroshchenko — CHANGES_REQUESTED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Review by DavidCharif — APPROVED** [review]

**Comment by ArtemLymarenko** [comment]
Added adaptivity
header-1
header-2
header-3

**Review by dimapopovych — APPROVED** [review]
It appears that the typography could be slightly off but could be fixed later, and I'm not sure if using CSS modules and @apply is necessary.

**Pull request merged**